Sources/CodeEditSourceEditor/Controller/TextViewController+IndentLines.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  TextViewController+IndentLines.swift
+//
+//
+//  Created by Ludwig, Tom on 11.09.24.
+//
+
+import CodeEditTextView
+import AppKit
+
+extension TextViewController {
+    /// Handels indentation and unindentation
+    ///
+    /// Handles the indentation of lines in the text view based on the current indentation option.
+    ///
+    /// This function assumes that the document is formatted according to the current selected indentation option.
+    /// It will not indent a tab character if spaces are selected, and vice versa. Ensure that the document is
+    /// properly formatted before invoking this function.
+    ///
+    /// - Parameter inwards: A Boolean flag indicating whether to outdent (default is `false`).
+    public func handleIndent(inwards: Bool = false) {
+        let indentationChars: String = indentOption.stringValue
+        guard !cursorPositions.isEmpty else { return }
+
+        textView.undoManager?.beginUndoGrouping()
+        for cursorPosition in self.cursorPositions {
```

Comment:
```suggestion
for cursorPosition in self.cursorPositions.reversed() {
```
After the text is updated from the first cursor, future cursor locations will be out of position by the size of the indent sometimes causing the wrong line to be indented. Reversing these fixes that.

https://github.com/user-attachments/assets/a8a353e5-eb85-468b-9572-36e1260eb244